rep-I0005.rst

```diff
@@ -0,0 +1,164 @@
+
+Simple Message is a joint-level interface that requires the external
+system to perform the inverse kinematics calculations. Robot
+controllers typically provide this function internally, and the
+proposed extensions enable external systems to benefit from the
+vendor's built-in kinematics. 
+
+The extensions also include messages for reporting the location and
+other information about objects in the robot's immediate environment
+that may be tracked using built-in capabilities of the robot
+controller. 
+
+Message Specifications
+======================
+
+Cartesian Move Request
+----------------------
+A Cartesian move message requests that the robot move to the specified
+Cartesian position and quaternion orientation along a straight line
+and about a fixed rotational axis. The translational speed applies to
+the motion along the straight line. The rotational speed applies to
+the motion about the fixed axis. Translational and rotational motion
+should finish at the same time, and in general one of these speeds may
+be slower than specified to ensure simultaneous arrival. The sequence
+number provided will be used in subsequent messages from the robot
+controller to associate replies and status with the move request.
+
+Format of a Cartesian move request::
+
+  int length: 4 bytes, constant value 13 x 4 = 52
+  int message_type: 4 bytes, constant value 31
+  int comm_type; 4 bytes, constant value 2, REQUEST
+  int reply_type: 4 bytes, ignored, any value, typically 0
+  int seq_number: 4 bytes, identifying this request
+  float x, y, z, qx, qy, qz, qw: 4 bytes each, for the requested
+  Cartesian position and quaternion rotation
+  float translational_speed: 4 bytes, the speed to be applied to the
+  translational part
+  float rotational_speed: 4 bytes, the speed to be applied to the
+  rotational part
+
+Cartesian Move Reply
+--------------------
+The robot controller should reply to each Cartesian move request with
+a reply that the request can be initiated. Immediately upon receipt of
+the move request, the robot controller should reply with this message
+with the reply_type set to 3 (EXECUTING), or 2 (FAILURE) if
+the move request is not possible. When the motion has finished, the
+robot controller should reply with the reply_type set to 1 (SUCCESS),
+or 3 (FAILURE) if the move did not finish where requested.
+
+Format of a Cartesian move reply::
+
+  int length: 4 bytes, constant value 4 x 4 = 16
+  int message_type: 4 bytes, constant value 31
+  int comm_type; 4 bytes, constant value 3, REPLY
+  int reply_type: 4 bytes, 1 = SUCCESS, 2 = FAILURE, 3 = EXECUTING
+  int seq_number: 4 bytes, identifying this request
+
+Cartesian Move State
+---------------------
+The robot controller should continually send back Cartesian move state
+messages as the move progresses, with the current pose and speeds of
+the robot. The format is the same as for the request message, except
+that the comm_type is 1, TOPIC. 
```

Comment:
Are implementations expected to treat *Cartesian Move State* messages as complementary to (fi) `JOINT_STATE` or `JOINT_FEEDBACK` messages (so should they be broadcast continuously)? Or only when the controller is performing a Cartesian motion?